code/ui/generalhud/nameplate/Nameplate.cs

```diff
@@ -0,0 +1,96 @@
+using Sandbox;
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+using TTTReborn.Player;
+using TTTReborn.Player.Camera;
+
+namespace TTTReborn.UI
+{
+    public class Nameplate : EntityHintPanel
+    {
+        public TTTPlayer Player;
+
+        private readonly Sandbox.UI.Panel _labelHolder;
+        private readonly Sandbox.UI.Panel _nameHolder;
+        private readonly Label _nameLabel;
+        private readonly Label _damageIndicatorLabel;
+
+        private struct HealthGroup
+        {
+            public string Title;
+            public Color Color;
+            public int MinHealth;
+
+            public HealthGroup(string title, Color color, int minHealth)
+            {
+                Title = title;
+                Color = color;
+                MinHealth = minHealth;
+            }
+        }
+
+        // Pay attention when adding new values! The highest health-based entry has to be the first item, etc.
+        private HealthGroup[] HealthGroupList = new HealthGroup[]
+        {
+            new HealthGroup("Healthy", Color.FromBytes(44, 233, 44), 66),
+            new HealthGroup("Injured", Color.FromBytes(233, 135, 44), 33),
+            new HealthGroup("Near Death", Color.FromBytes(252, 42, 42), 0)
+        };
+
+        public Nameplate(TTTPlayer player) : base()
+        {
+            Player = player;
+
+            StyleSheet.Load("/ui/generalhud/nameplate/Nameplate.scss");
+
+            AddClass("text-shadow");
+
+            _labelHolder = Add.Panel("label-holder");
+
+            _nameHolder = _labelHolder.Add.Panel("name-holder");
+            _nameLabel = _nameHolder.Add.Label("", "name");
+
+            _damageIndicatorLabel = _labelHolder.Add.Label("", "damage-indicator");
+
+            Enabled = false;
+        }
+
+        private HealthGroup GetHealthGroup(float health)
+        {
+            foreach (HealthGroup healthGroup in HealthGroupList)
+            {
+                if (health >= healthGroup.MinHealth)
+                {
+                    return healthGroup;
+                }
+            }
+
+            return HealthGroupList[HealthGroupList.Length - 1];
```

Comment:
```suggestion
            return HealthGroupList[^1];
```

Reply:
> Additionally, you added the file `materials/ammo/9mmbox.vmat_c `, intended?

The vmats are the worst. They appear sometimes and when making a lot of changes they just slip by and end up in the commit. I'm just glad only one made it through given that I had so many commits made.